### PR TITLE
Slab geojson

### PR DESCRIPTION
**Pull request by kirstybayliss**
This PR adds the option to create profiles for subduction using a geojson file containing manually chosen cross-sections, the dip direction and length. The get_profiles_geojson function fits into the subduction workflow in place of the get_profiles function inside the call to get_profiles_from_slab2pt0, or get_profiles_from_slab2pt0_geojson can be called directly in the workflow as a replacement in the subduction profile, specifying the geojson file in the toml. 

Also fixes a minor weirdness with the interpolator, where it would previously fail occasionally for cross-sections that trended along a single x or y coordinate. Instead we call the tesselation step directly and allow it to add a small jitter to account for this if the original tesselation fails (otherwise there will be no change). If this is still failing for a provided geometry, no cross-sections will be generated.

Tests imply that the difference in depths when using the 'jitter' are less than 0.03km. 
image

Updated so that the old version is used unless we have an issue like this, in which case we assume that a small difference in depth is acceptable.

### CODE REVIEW

**Comment by kirstybayliss** [comment]
Hmm actually the classification step is giving the dreaded 'Surface does not conform with Aki & Richards' error, even though the 3D plot and cross-sections look fine. Investigating!

**Comment by kirstybayliss** [comment]
- [ ] Add test for full suduction classification workflow

File: openquake/sub/cross_sections.py [hunk 1]
```diff
@@ -113,11 +113,12 @@ def compute_profiles(self, bffer):
 
             # Get min and max longitude and latitude values
             minlo, maxlo, minla, maxla, qual = cs.get_mm(2.0)
-
+            
+            #Sbreakpoint()
```
Comment (mmpagani) [inline openquake/sub/cross_sections.py:117]: leftover

File: openquake/sub/tests/create_profiles_from_slab2pt0_test.py [hunk 2]
```diff
@@ -31,14 +31,16 @@
...
+                                                      get_profiles_geojson)
 from openquake.sub.cross_sections import CrossSection, Slab2pt0
 
+
 pygmt_available = False
 try:
     import pygmt
-except ImportError:
     pygmt_available = True
+except ImportError:
```
Comment (mmpagani) [inline openquake/sub/tests/create_profiles_from_slab2pt0_test.py:43]: maybe here we want to set 'pygmt_available' as False

File: openquake/sub/tests/data/depthtomoho.xyz
Comment (mmpagani) [inline openquake/sub/tests/data/depthtomoho.xyz:1]: Can we upload this file? No licence issues?

File: openquake/sub/tests/profile_workflow_classification_test.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	os.makedirs(sfc_sl)
+
+	build_complex_surface(sfc_out, max_sampl_dist, sfc_in,
+                              upper_depth_int, lower_depth_int)
+
+	build_complex_surface(sfc_out, max_sampl_dist, sfc_sl,
+                              lower_depth_int, lower_depth_slab)
+
+
+	classify.classify(classification_ini, True)
```
Comment (mmpagani) [inline openquake/sub/tests/profile_workflow_classification_test.py:59]: I guess this test executes the workflow but I do not see checks at the end that the results are as expected. If we cannot define what's expected, we should at least save the current results to check that the code returns the same information.

**Review by mmpagani — APPROVED** [review]
Many thanks Kirsty! I left a few minor comments. Otherwise the PR looks good to me.

File: openquake/sub/tests/data/depthtomoho.xyz
Comment (kirstybayliss) [reply openquake/sub/tests/data/depthtomoho.xyz:1]: Doesn't seem to have a license, added a reference to the web download and the REM model in the test

File: openquake/sub/tests/profile_workflow_classification_test.py (hunk 3 above)
Comment (kirstybayliss) [reply openquake/sub/tests/profile_workflow_classification_test.py:67]: Now tests that the classification results are as expected

**Pull request merged**